contracts/bond/CollateralWhitelist.sol

```diff
@@ -67,22 +79,34 @@ abstract contract CollateralWhitelist is Initializable {
 
         string memory symbol = IERC20MetadataUpgradeable(erc20CollateralTokens)
             .symbol();
-        require(isCollateralWhitelisted(symbol), "Whitelist: not whitelisted");
         require(
-            _whitelist[symbol] != erc20CollateralTokens,
-            "Whitelist: identical address"
+            isCollateralWhitelisted(erc20CollateralTokens),
+            "Whitelist: not whitelisted"
+        );
+
+        bytes32 storedSymbolHash = keccak256(
+            abi.encode(_symbols[erc20CollateralTokens])
+        );
+        bytes32 symbolHash = keccak256(abi.encode(symbol));
+
+        require(storedSymbolHash != symbolHash, "Whitelist: same symbol");
```

Comment:
question: does the logic still match the function name & description?

Originally this function was updating the key part of the `symbol=>address` mapping. 
Now we're dealing with addresses rather than symbols, could we ditch this update function entirely 🤔 

As long as there's an add & remove the collateral token address, that would suffice.

Reply:
I agree it can be removed!